### DISCUSSION TITLE
why numeric key is parse as StringNode

### DISCUSSION THREAD

**Discussion by zhuliquan** [antonmedv/expr#394] [General]
https://github.com/antonmedv/expr/blob/d2100ece96affe6766d97e2cf0e4ca4d145f0d30/parser/parser.go#L454C32-L454C32
I am confused why numeric key is parse as `ast.StringNode`

**Comment by antonmedv** [comment]
Maps created in expr runtime can only be with string keys. map[string]any

**Comment by zhuliquan** [comment]
oh, I got it. but I think you should give instructions about map in the docs (https://github.com/zhuliquan/expr/blob/feature-reuse_common/docs/Language-Definition.md), otherwise someone may use result of `vm.Run` as `map[int]interface{}`
